dataplane/server.go

```diff
@@ -24,3 +35,57 @@ const (
 type Server struct {
 	dpb.UnimplementedHALServer
 }
+
+var (
+	// Stubs for testing
+	setInterfaceHWAddr = kernel.SetInterfaceHWAddr
+	setInterfaceIPs    = kernel.SetInterfaceIPs
+	setInterfaceState  = kernel.SetInterfaceState
+)
+
+// UpdatePort updates an interface properties from the input request.
+func (s *Server) UpdatePort(_ context.Context, req *dpb.UpdatePortRequest) (*dpb.UpdatePortResponse, error) {
+	if req.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "name not set")
+	}
+	tapName := fmt.Sprintf("%s-tap", req.Name)
```

Comment:
Should this name conversion be a helper function?

Reply:
will do once needed elsewhere